src/main/java/pl/edu/agh/firecell/engine/algorithm/TemperatureCalculation.java

```diff
@@ -0,0 +1,165 @@
+package pl.edu.agh.firecell.engine.algorithm;
+
+import org.joml.Vector3i;
+import pl.edu.agh.firecell.model.Cell;
+import pl.edu.agh.firecell.model.Material;
+import pl.edu.agh.firecell.model.MatterState;
+import pl.edu.agh.firecell.model.State;
+import pl.edu.agh.firecell.model.util.NeighbourUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static pl.edu.agh.firecell.engine.algorithm.BasicAlgorithm.CONDUCTIVITY_COEFFICIENT;
+import static pl.edu.agh.firecell.engine.algorithm.BasicAlgorithm.CONVECTION_COEFFICIENT;
+
+public class TemperatureCalculation {
```

Comment:
`TemperaturePropagator` will be better name, and `FirePropagator` for the fire propagation